src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -687,26 +689,64 @@ internal void TogglePreviewControlAllowance()
 
         #endregion
 
-        private void OptionsButton_Click(object sender, RoutedEventArgs e)
+        /// <summary>
+        /// A dictionary of MenuItems which are added during certain nodes' NodeViewCustomization process.
+        /// </summary>
+        private Dictionary<string, object> NodeViewCustomizationMenuItems { get; } = new Dictionary<string, object>();
```

Comment:
why are the values of type `object` and not `MenuItem`?

Reply:
@OliverEGreen a few other issues with this...


Do you need the order of keys / values in this dictionary to stay stable? If that is the case you need to use an ordered dictionary like:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.specialized.ordereddictionary?view=netframework-4.8

also I was curious why you did not just use a hashset - it looks like MenuItem's base class ItemControl implements `Equals`
https://referencesource.microsoft.com/#PresentationFramework/src/Framework/System/windows/Controls/ItemsControl.cs,3808

though - if you need order, go with the ordered dictionary, hashset does not preserve insertion order either.

Reply:
"why are the values of type object and not MenuItem?"

As above, they can be both Separators and MenuItems.

Reply:
I'd say order stability does matter yeah, will switch out to an OrderedDictionary! 

I considered HashSet, but I wasn't sure about using Equals to compare one MenuItem to another - would that work? Hence, I'm using the 'Header' values to indicate uniqueness.